R/uniDimFrequentistPreCalc.R

```diff
@@ -0,0 +1,141 @@
+
+.frequentistPreCalc <- function(jaspResults, dataset, options) {
+
+  if (!is.null(.getStateContainerF(jaspResults)[["modelObj"]]$object)) {
+    if (!is.null(.getStateContainerF(jaspResults)[["modelObj"]]$object$bootSamp))
+      return(.getStateContainerF(jaspResults)[["modelObj"]]$object)
+  }
+
+
+
+  derivedOptions <- .frequentistDerivedOptions(options)
+
+  # what if no coefficient boxes are checked?
+  if(!any(derivedOptions[["selectedEstimators"]]) & !any(derivedOptions[["itemDroppedSelected"]])) {
+    variables <- options[["variables"]]
+    # if (length(options[["reverseScaledItems"]]) > 0L) {
+    #   dataset <- .reverseScoreItems(dataset, options)
+    # }
+    empty <-  T
+    model <- list(empty = empty)
+    model[["footnote"]] <- .checkLoadings(dataset, variables)
+    return(model)
+  }
+
+  # what if too few variables are entered:
+  if (length(options[["variables"]]) < 3) {
+    empty <-  T
+    model <- list(empty = empty)
+    model[["footnote"]] <- gettext("Please enter at least 3 Variables to do an analysis")
+    return(model)
+  }
+
+  model <- jaspResults[["modelObj"]]$object
+
+  if (is.null(model)) {
+    model <- list()
+    # check for missings and determine the missing handling
+    if (anyNA(dataset)) {
+      if (options[["missingValues"]] == "excludeCasesPairwise") {
+        model[["use.cases"]] <- "pairwise.complete.obs"
+        model[["pairwise"]] <- TRUE
+        model[["footnote"]] <- gettextf("%s Of the observations, pairwise complete cases were used. ",
+                                        model[["footnote"]])
+      } else {
+        pos <- which(is.na(dataset), arr.ind = TRUE)[, 1]
+        dataset <- dataset[-pos, ]
```

Comment:
you could use `complete.cases` here, but I would use the `exclude.na.listwise` when you read in the dataset:
```r
.readDataSetToEnd(columns.as.numeric = variables, columns.as.factor = NULL, exclude.na.listwise = NULL)
```

Reply:
Ok. What would happen though, when I have read in the data with listwise deletion, but in the analysis I then change it to pairwise deletion. Would the dataset then be re-read again?

Reply:
well, yes but isn't that always the case?

Reply:
that is news to me. But I guess that is my mistake.